src/storage/page_buffer.c

```diff
@@ -4920,7 +4920,8 @@ pgbuf_is_temporary_volume (VOLID volid)
     {
       return false;
     }
-  return (LOG_DBFIRST_VOLID <= volid && xdisk_get_purpose (NULL, volid) == DB_TEMPORARY_DATA_PURPOSE);
+  assert (LOG_DBFIRST_VOLID <= volid);
```

Comment:
Considering the purpose of this function, it seems safe to remove assert.

Reply:
It's corrected.

Reply:
The range of VOLID is not only positive. Auxiliary files have negative values and it can be given if a user(developer) uses this function wrongly. So, it seems to me that the assertion works as a pre-condition for this function.

Reply:
@wrlawodms The condition 'LOG_DBFIRST_VOLID <= volid' is checked in the xdisk_get_purpose() function by if condition. Since the pgbuf_is_temporary_volume() function only needs to determine the purpose of temp regardless of whether the volid is positive or negative, it is considered safe to remove the assert.

Reply:
@hornetmj OK. Now, I agree. Thanks for the elaboration.